esp-riscv-rt/src/lib.rs

```diff
@@ -632,44 +636,8 @@ _start_trap31:
     sw ra, 0(sp)
     la ra, cpu_int_31_handler
     j _start_trap_direct
-"#,
-#[cfg(not(feature="direct-vectoring"))]
-r#"
-_start_trap1:
-_start_trap2:
-_start_trap3:
-_start_trap4:
-_start_trap5:
-_start_trap6:
-_start_trap7:
-_start_trap8:
-_start_trap9:
-_start_trap10:
-_start_trap11:
-_start_trap12:
-_start_trap13:
-_start_trap14:
-_start_trap15:
-_start_trap16:
-_start_trap17:
-_start_trap18:
-_start_trap19:
-_start_trap20:
-_start_trap21:
-_start_trap22:
-_start_trap23:
-_start_trap24:
-_start_trap25:
-_start_trap26:
-_start_trap27:
-_start_trap28:
-_start_trap29:
-_start_trap30:
-_start_trap31:
-
-"#,
-r#"
-_start_trap:
+la ra, _start_trap_rust_hal /* this runs on exception, use regular fault handler */
```

Comment:
I'm not _100_ percent sure of the purpose of this line, @onsdagens do you recall?

Reply:
we want the ISR address in RA, [this line](https://github.com/esp-rs/esp-hal/blob/main/esp-riscv-rt/src/lib.rs#L767C1-L767C54) uses it to jump into the handler. This line should only run on exception, and loads the default handler address into RA.

Reply:
the default being the `vectored` handler i believe.

Reply:
Gotcha, thanks for clarifying!